include/nlohmann/adl_serializer.hpp

```diff
@@ -20,8 +20,10 @@ struct adl_serializer
     @param[in,out] val  value to write to
     */
     template<typename BasicJsonType, typename ValueType>
-    static void from_json(BasicJsonType&& j, ValueType& val) noexcept(
-        noexcept(::nlohmann::from_json(std::forward<BasicJsonType>(j), val)))
+    static auto from_json(BasicJsonType&& j, ValueType& val) noexcept(
+        noexcept(::nlohmann::from_json(std::forward<BasicJsonType>(j), val))) -> decltype(
+            ::nlohmann::from_json(std::forward<BasicJsonType>(j), val)
+        )
```

Comment:
You need to add a `return` here or it will be malformed if the function you call has a return type.

Reply:
In this case `::nlohmann::from_json` returns `void`, but I should add `, void()` in the decltype to be extra sure.

Reply:
Why do you need the `decltype` at all?  Is that for SFINAE?  Is that not covered by the expression in the `noexcept`?

Reply:
I thought so at first, but doing a qualified call inside a `noexcept` breaks. The `decltype` is mandatory here.

Reply:
Okay, thanks.